homeassistant/components/google/api.py

```diff
@@ -0,0 +1,179 @@
+        """
+        List events.
+
+        https://developers.google.com/calendar/v3/reference/events/list
+        """
+        # TODO: List events for this calendar month # pylint: disable=fixme
+        # TODO: List events for the next 7 days # pylint: disable=fixme
+        # TODO: List events for today # pylint: disable=fixme
+        # TODO: List x scheduled/upcoming events # pylint: disable=fixme
+        date_start = (
+            datetime.datetime.utcnow().isoformat() + "Z"
+        )  # 'Z' indicates UTC time
+        date_end = (
+            datetime.datetime.utcnow().isoformat() + "Z"
+        )  # 'Z' indicates UTC time
+        events = (
+            self._api.events()  # pylint: disable=maybe-no-member
+            .list(
+                calendarId=calendar_id,
+                timeMin=date_start,
+                timeMax=date_end,
+                maxResults=count,
+                singleEvents=True,
+                orderBy="startTime",
+            )
+            .execute()
+        )
+
+        return events.get("items", [])
+
+
+class GooglePeopleAPI:
+    """
+    Google People API.
+
+    https://developers.google.com/people
+    http://googleapis.github.io/google-api-python-client/docs/dyn/people_v1.people.html
+    """
+
+    def __init__(self, creds: Credentials):
+        """Initialize Google People API."""
+        self._api = build("people", "v1", credentials=creds, cache_discovery=False)
+
+    def list_contacts(self, count=None) -> List:
+        """
+        List contacts.
+
+        https://developers.google.com/people/api/rest/v1/people.connections/list
+        """
+        contacts = []
+        page_token = ""
+        while True:
+            results = (
+                self._api.people()  # pylint: disable=maybe-no-member
+                .connections()
+                .list(
+                    sortOrder="LAST_NAME_ASCENDING",
+                    resourceName="people/me",
+                    pageSize=count,
+                    personFields="birthdays,events,names",
+                    pageToken=page_token,
+                )
+                .execute()
+            )
+            contacts.extend(results.get("connections", []))
+
+            page_token = results.get("nextPageToken")
+            if not page_token:
+                break
+
+        return contacts
+
+
+class ConfigEntryAuth(GoogleAPI):
+    """Provide authentication tied to an OAuth2 based config entry."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        entry: ConfigEntry,
+        implementation: AbstractOAuth2Implementation,
+    ):
+        """Initialize API ConfigEntryAuth."""
+        self.hass = hass
+        self.config_entry = entry
+        self.session = OAuth2Session(hass, entry, implementation)
+        client_id = entry.data.get(CONF_CLIENT_ID)
+        client_secret = entry.data.get(CONF_CLIENT_SECRET)
+
+        super().__init__(client_id, client_secret, self.session.token)
```

Comment:
The super constructor writes to `session.token`, so you need to make a copy before pasting it in ? 

There is no OAuth2 standard that requires the client ID / secret to be in the token. Can you link me to the docs that states this ?